src/modules/product/interface/product.routes.ts

```
@@ -30,6 +30,44 @@ productRouter.get('/store/:storeId', (req, res, next) =>
+  '/:id/image',
+  uploadProductImage,
+  (req, res, next) => productController.uploadImage(req, res, next)
+);
+
+// Create new product with image
+// Helper function to handle file upload and product data
+const handleProductUpload = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+  controllerMethod: Function
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)